### ISSUE TITLE
var fields of a module not accessible from outside

### ISSUE THREAD

**Issue by JonathanAldrich** [wyvernlang/wyvern#170]
There seems to be some issues with this - need to investigate.  Sample test that fails: try to have cellModuleClient access the value field of cellAsModule (after removing the ascription that should hide it).

**Comment by Darya-** [comment]
Isn't this intended behavior? In our modules paper, we limit access to var fields from the outside forcing the programmer to use getters and setters.

**Comment by JonathanAldrich** [comment]
Yes, although I think things are actually a bit more subtle -- var fields of a module that are not hidden are internally converted into a pair of getters and setters, which can be called from outside (but using syntax that looks like accessing the var field).  This design decision could probably be fleshed out better.

**Comment by JonathanAldrich** [comment]
Update: it looks like the getters/setters weren't being exported from a module.  As of a commit from today, they are, but you can only access them by using the internal name (e.g. m._getA() instead of m.a)

So this issue would be resolved by having "x.a" do a search in the type for an appropriately-named getter (or setter, in the case of assignment).

Or, we can decide we don't want to ever support this.  But in that case, it would be nice for the compiler to output some kind of warning message when people try to do it, because this is otherwise a "gotcha" for new people to Wyvern.

**Comment by Darya-** [comment]
`var` fields are absent from types, and so I think that `var` fields of a module (and, for consistency, of an object too) should not be accessible from outside the module (or the object) unless the programmer adds getters and setters for them. Allowing access to `var` fields through the internal name seems to be unsafe, and so I think this should not be allowed either.

It would indeed be nice to have an informative error message when the programmer tries to access a `var` field from the outside of a module (or object).
